utils/python/CIME/case_st_archive.py

```diff
@@ -1,24 +1,413 @@
+    logfiles = glob.glob(os.path.join(rundir,'*.log.*'))
+    for logfile in logfiles:
+        srcfile = join(rundir, os.path.basename(logfile))
+        destfile = join(archive_logdir, os.path.basename(logfile))
+        shutil.move(srcfile, destfile)
+
+
+###############################################################################
+def archive_history_files(case, archive, archive_entry,
+                          compclass, compname, histfiles_savein_rundir):
+###############################################################################
+
+    # determine history archive directory (create if it does not exist)
+    dout_s_root = case.get_value("DOUT_S_ROOT")
+    archive_histdir = os.path.join(dout_s_root, compclass, 'hist')
+    if not os.path.exists(archive_histdir):
+        os.makedirs(archive_histdir)
+        logger.debug("created directory %s" %archive_histdir)
+
+    # determine ninst and ninst_string
+    ninst, ninst_string = get_ninst_info(case, compclass)
+
+    # archive history files - the only history files that kept in the
+    # run directory are those that are needed for restarts
+    rundir = case.get_value("RUNDIR")
+    for suffix in archive.get_hist_file_extensions(archive_entry):
+        for i in range(ninst):
+            if ninst_string:
+                newsuffix = compname + ".*" + ninst_string[i] + suffix
+            else:
+                newsuffix = compname + ".*"  + suffix
+            logger.debug("short term archiving suffix is %s " %newsuffix)
+            p = re.compile(newsuffix)
+            histfiles = [ f for f in os.listdir(rundir) if p.search(f) ]
+            if histfiles:
+                logger.debug("hist files are %s " %histfiles)
+                for histfile in histfiles:
+                    srcfile = join(rundir, histfile)
+                    expect(os.path.isfile(srcfile),
+                           "history file %s does not exist " %srcfile)
+                    destfile = join(archive_histdir, histfile)
+                    if histfile in histfiles_savein_rundir:
+                        logger.debug("copying %s to %s " %(srcfile, destfile))
+                        shutil.copy(srcfile, destfile)
+                    else:
+                        logger.debug("moving %s to %s " %(srcfile, destfile))
+                        shutil.move(srcfile, destfile)
+
+
+###############################################################################
+def get_histfiles_for_restarts(case, archive, archive_entry, restfile):
+###############################################################################
+
+    # determine history files that are needed for restarts 
+    histfiles = []
+    rest_hist_varname = archive.get_entry_value('rest_history_varname', archive_entry)
+    if rest_hist_varname != 'unset':
+        rundir = case.get_value("RUNDIR")
+        cmd = "ncdump -v %s %s " %(rest_hist_varname, os.path.join(rundir,restfile))
+        rc,out,error = run_cmd(cmd, ok_to_fail=True)
```

Comment:
It looks like you ignored the return code. Should we be checking that it succeeded?